lib/experimental/modules/class-gutenberg-modules.php

```diff
@@ -262,13 +262,14 @@ function gutenberg_dequeue_module( $module_identifier ) {
 	Gutenberg_Modules::dequeue( $module_identifier );
 }
 
-// Prints the import map in the head tag.
-add_action( 'wp_head', array( 'Gutenberg_Modules', 'print_import_map' ) );
+$modules_position = wp_is_block_theme() ? 'wp_head' : 'wp_footer';
+// Prints the import map in the head tag in block themes. Otherwise in the footer.
+add_action( $modules_position, array( 'Gutenberg_Modules', 'print_import_map' ) );
 
-// Prints the enqueued modules in the head tag.
-add_action( 'wp_head', array( 'Gutenberg_Modules', 'print_enqueued_modules' ) );
+// Prints the enqueued modules in the head tag in block themes. Otherwise in the footer.
+add_action( $modules_position, array( 'Gutenberg_Modules', 'print_enqueued_modules' ) );
 
-// Prints the preloaded modules in the head tag.
+// Prints the preloaded modules in the head tag in block themes.
 add_action( 'wp_head', array( 'Gutenberg_Modules', 'print_module_preloads' ) );
```

Comment:
I'm not well versed in the modules code now, but it seems like this may be an issue. Won't all the module loading rely on import maps? In that case, wouldn't we need to move all the module loading (including the "preload" modules) to the footer?

Reply:
Let's prevent it by moving all scripts to the footer.